### DISCUSSION TITLE
wgpu example

### DISCUSSION THREAD

**Discussion by Bastl34** [emilk/egui#2432] [Q&A]
Is there a working exmaple for wgpu? I mean based on the current/latest egui version.
I have checked this: https://github.com/hasenbanck/egui_example
But it still relies on `egui_wgpu_backend` not on `egui-wgpu` and it is not working if you upgrade to  `egui-wgpu`

**Comment by John-Nagle** [comment]
I have one at ui-mock on Github, but it's not for the latest egui. I'm using Rend3, and it's still bound to an older egui.

**Reply by cfagot** [reply]
Do you have a link for this? I checked out your ui-mock project but that doesn't use equi-wgpu.

Also does anyone know where to find the history of equi-wpgu before it was imported into the equi project. I'm so confused about this. The change log mentions a project that doesn't look at all like the current equi-wgpu:

https://github.com/LU15W1R7H/eww

On the other hand, the readme mentions a project which is very similar (and is the one I am porting from):

https://github.com/hasenbanck/egui_wgpu_backend

It is very similar but there seems to be a significant gap in history (e.g., can't see when add_textures disappeared or what the replacement was).

A simple integration example would sure clear a lot up.

**Comment by haydn-jones** [comment]
This would also be very useful to me.

**Comment by John-Nagle** [comment]
Right now, working on bringing up ui-mock to current versions of everything. 

Current issue:

```
   Compiling rend3 v0.3.0 (https://github.com/BVE-Reborn/rend3.git#cd6c6438)
error[E0560]: struct `Limits` has no field named `max_bindings_per_bind_group`
  --> /home/john/.cargo/git/checkouts/rend3-e03f89403de3386a/cd6c643/rend3/src/setup.rs:85:5
   |
85 |     max_bindings_per_bind_group: 640,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Limits` does not have this field
   |
   = note: available fields are: `max_texture_dimension_1d`, `max_texture_dimension_2d`, `max_texture_dimension_3d`, `max_texture_array_layers`, `max_bind_groups` ... and 23 others
```

**Comment by John-Nagle** [comment]
> Do you have a link for this? I checked out your ui-mock project but that doesn't use equi-wgpu.

Oh, right. I'm using egui, and wgpu, but not egui-wgpu. I'm using Rend3, which uses rend3-egui as the glue layer.

Some assembly required.

**Reply by cfagot** [reply]
That makes sense. Thanks.

**Comment by haydn-jones** [comment]
I figured it out, if you give me a few days I can put together an example (pretty busy rn). 

Edit: I have it working in one of my repos but its pretty damn big so a small example will be more illustrative.

**Comment by cfagot** [comment]
Oh, just noticed that eframe optionally uses egui-wgpu now, so it provides an example of how to use egui-wgpu (it also uses egui-winit so it is an example of that too). Still wish we had the revision history from the last version of the source project to now, but working example is the main thing.

**Reply by Gardene-el** [reply]
hi, I'm wondering where `an example of how to use egui-wgpu` is, since I can't find it on official [examples](https://github.com/emilk/egui/tree/master/examples). 
:)

**Reply by Norlock** [reply]
https://github.com/Norlock/sparticles-egui/blob/main/src/model/gfx_state.rs. You can use mine, its a bit to big to be used as an official example but you can look at draw_gui is kind of everything you need. Its being rendered here at fn render: https://github.com/Norlock/sparticles-egui/blob/main/src/fx/post_process.rs. If you need more help you can ask questions on the WGPU matrix, most people there can help you out or me if I'm there. 

However this is not including eframe.

**Comment by John-Nagle** [comment]
Can't help with current versions. I'm stuck waiting for arcanization to land.
